File: Datasets/TrickyBugs/GenProgs/dpp_generated_progs_python/p02943/p02943_num8_parsed.py
# Let's generate the python program to achieve the given intention

# Read the inputs
N, K = map(int, input().split())
S = input()

# Perform the operations K times
for _ in range(K):
    T = S[::-1]  # Reverse the string
    U = S + T  # Concatenate S and T
    S = min(U[i:i+N] for i in range(len(S)+1-N))  # Find lexicographically smallest substring of U with length N

# Print the resulting string
print(S)

File: Datasets/TrickyBugs/GenProgs/dpp_generated_progs_cpp/p02943/p02943_num8_parsed.cpp
#include <iostream>
#include <string>
#include <algorithm>

int main() {
    int N, K;
    std::string S;
    std::cin >> N >> K >> S;
    
    // Reverse S
    std::string T = S;
    std::reverse(T.begin(), T.end());
    
    // Concatenate S and T
    std::string U = S + T;
    
    // Find the lexicographically smallest substring of length N from U
    std::string smallest = U.substr(0, N);
    
    // Perform the operation K times
    for (int i = 1; i < K; i++) {
        // Shift the lexicographically smallest substring to the right
        std::rotate(smallest.begin(), smallest.begin() + 1, smallest.end());
        
        // Update the lexicographically smallest substring if a smaller one is found
        std::string current = U.substr(i, N);
        smallest = std::min(smallest, current);
    }
    
    // Print the lexicographically smallest possible string after K operations
    std::cout << smallest << std::endl;
    
    return 0;
}